### PR TITLE
23 improve the error handling support for the signature, core & registry resolver crates

### PR DESCRIPTION
**Pull request by LuisOsta**

### CODE REVIEW

**Review by sKudryashov — CHANGES_REQUESTED** [review]

File: registry_resolver/src/lib.rs [hunk 1]
```diff
@@ -192,13 +183,13 @@ mod tests {
             document: None,
             metadata: None,
          }))),
-        Some(ssi_core::error::ErrorKind::DocumentNotFound),
+        Some(ssi_core::error::ResolverError::DocumentNotFound("No document found associated with did:knox:z6MkfFmsob7fC3MmqU1JVfdBnMbnAw7xm1mrEtPvAoojLcRh".to_string())),
```
[the remark this answers is not in the document]
Comment (LuisOsta) [reply registry_resolver/src/lib.rs:186]: Yes that's the hard-coded did defined in `create_did` but in general I'm hoping to improve the testing situation in the registry resolver in a follow up PR

File: registry_resolver/src/lib.rs [hunk 2]
```diff
@@ -119,7 +108,7 @@ mod tests {
         create_did(),
         create_did_doc(create_did()),
         Some(Err(tonic::Status::invalid_argument("message"))),
-        Some(ssi_core::error::ErrorKind::NetworkFailure),
+        Some(ssi_core::error::ResolverError::NetworkFailure("status: InvalidArgument, message: \"message\", details: [], metadata: MetadataMap { headers: {} }".to_string())),
```
[the remark this answers is not in the document]
Comment (LuisOsta) [reply registry_resolver/src/lib.rs:111]: What's the benefit of a raw string literal?

File: registry_resolver/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (LuisOsta) [reply registry_resolver/src/lib.rs]: @sKudryashov Can I resolve this?

[the remark this answers is not in the document]
Comment (sKudryashov) [reply registry_resolver/src/lib.rs]: yes

**Review by sKudryashov — APPROVED** [review]

**Pull request merged**